mesonbuild/modules/gnome.py

```diff
@@ -402,6 +400,8 @@ def generate_gir(self, state, args, kwargs):
                 self.gir_dep = PkgConfigDependency('gobject-introspection-1.0',
                                                    state.environment,
                                                    {'native': True})
+            giscanner = os.environ['PKG_CONFIG_SYSROOT_DIR'] + self.gir_dep.get_pkgconfig_variable('g_ir_scanner', {})
+            gicompiler = os.environ['PKG_CONFIG_SYSROOT_DIR'] + self.gir_dep.get_pkgconfig_variable('g_ir_compiler', {})
```

Comment:
See e.g. #4380

This now respects pkg-config in the general sense, which should indirectly mean it supports the sysroot dir.